pkg/csi/service/common/authmanager.go

```diff
@@ -63,58 +73,96 @@ func GetAuthorizationService(ctx context.Context, vc *cnsvsphere.VirtualCenter)
-	datastoreIgnoreMapForBlockVolumes := make(map[string]*cnsvsphere.DatastoreInfo)
+// GetDatastoreMapForBlockVolumes returns a DatastoreMapForBlockVolumes. This map maps datastore url to datastore info
+// which need to be used when creating block volume.
+func (authManager *AuthManager) GetDatastoreMapForBlockVolumes(ctx context.Context) map[string]*cnsvsphere.DatastoreInfo {
+	datastoreMapForBlockVolumes := make(map[string]*cnsvsphere.DatastoreInfo)
 	authManager.rwMutex.RLock()
 	defer authManager.rwMutex.RUnlock()
-	for dsURL, dsInfo := range authManager.datastoreIgnoreMapForBlockVolumes {
-		datastoreIgnoreMapForBlockVolumes[dsURL] = dsInfo
+	for dsURL, dsInfo := range authManager.datastoreMapForBlockVolumes {
+		datastoreMapForBlockVolumes[dsURL] = dsInfo
+	}
+	return datastoreMapForBlockVolumes
+}
+
+// GetDatastoreMapForFileVolumes returns a DatastoreMapForFileVolumes. This map maps datastore url to datastore info
+// which need to be used when creating file volume.
+func (authManager *AuthManager) GetDatastoreMapForFileVolumes(ctx context.Context) map[string]*cnsvsphere.DatastoreInfo {
+	datastoreMapForFileVolumes := make(map[string]*cnsvsphere.DatastoreInfo)
+	authManager.rwMutex.RLock()
+	defer authManager.rwMutex.RUnlock()
+	for dsURL, dsInfo := range authManager.datastoreMapForFileVolumes {
+		datastoreMapForFileVolumes[dsURL] = dsInfo
+	}
+	return datastoreMapForFileVolumes
+}
+
+// refreshDatastoreMap scans all datastores in vCenter to check privileges, and compute the
+// datastoreMapForBlockVolumes
+func (authManager *AuthManager) refreshDatastoreMapForBlockVolumes() {
+	ctx, log := logger.GetNewContextWithLogger()
+	log.Info("auth manager: refreshDatastoreMapForBlockVolumes is triggered")
+	newDatastoreMapForBlockVolumes, err := GenerateDatastoreMapForBlockVolumes(ctx, authManager.vcenter)
+	if err == nil {
+		authManager.rwMutex.Lock()
+		defer authManager.rwMutex.Unlock()
+		authManager.datastoreMapForBlockVolumes = newDatastoreMapForBlockVolumes
+		log.Debugf("auth manager: datastoreMapForBlockVolumes is updated to %v", newDatastoreMapForBlockVolumes)
+	} else {
+		log.Warnf("auth manager: failed to get updated datastoreMapForBlockVolumes, Err: %v", err)
 	}
-	return datastoreIgnoreMapForBlockVolumes
 }
 
-// refreshDatastoreIgnoreMap scans all datastores in vCenter to check privileges, and compute the DatastoreIgnoreMap
-func (authManager *AuthManager) refreshDatastoreIgnoreMap() {
+// refreshDatastoreMapForFileVolumes scans all vSAN datastores with vSAN FS enabled in vCenter to
+// check privileges, and compute the datastoreMapForFileVolumes
+func (authManager *AuthManager) refreshDatastoreMapForFileVolumes() {
 	ctx, log := logger.GetNewContextWithLogger()
-	log.Info("auth manager: refreshDatastoreIgnoreMap is triggered")
-	newDatastoreIgnoreMapForBlockVolumes, err := GenerateDatastoreIgnoreMapForBlockVolumes(ctx, authManager.vcenter)
+	log.Info("auth manager: refreshDatastoreMapForFileVolumes is triggered")
```

Comment:
How often is this logged ? Wouldn't this log message flood the controller logs.

Reply:
moved to debug.